### DISCUSSION TITLE
How to get target branch when pull request

### DISCUSSION THREAD

**Discussion by ROBOTIS-David** [community/community#25298] [Actions]
<p>Hi, all</p>
<p>First of all, thank you so much for your time on my ticket.</p>
<p>Let’s say, I am working on ‘develop’ branch, and I got an PR from someone else,</p>
<p>I am trying to retrieve my target branch in ci file during the workflow run. But it only returns ‘refs/pull/36/merge’ although I use <code>${{github.base_ref}}</code>.</p>
<p>I am pretty sure <code>${{github.base_ref}}</code> is wrongly used.</p>
<p>May I ask any advice on how to the target branch <code>refs/heads/develop</code>, not <code>refs/pull/36/merge</code> if PR occurs?</p>
<p>Here is excerption of my workflow file.</p>
<pre><code class="lang-auto">jobs:
  my_ci:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
    steps:
      - uses: actions/checkout@v2
      - name: Run Arduino CI Script
        if: ${{github.ref}} == 'refs/heads/master' || ${{github.ref}} == 'refs/heads/develop'
        run: |
          source &lt;(my_source_url) ${{github.ref}}
      - name: Run Arduino CI Script (PR)
        run: |
          source &lt;(curl -SLs my_source_url) ${{github.base_ref}}
</code></pre>

**Comment by jsoref** [comment, marked as answer]
<p>So, I set up a thing for this:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/jsoref/community-events/runs/5170886359?check_suite_focus=true#step:2:46">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/jsoref/community-events/runs/5170886359?check_suite_focus=true#step:2:46" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/2119212/181090140-818df1a3-13a3-42a0-b8df-a6bd36d50b48.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/a/4a49215e09f63680f6bf0999a21ce4a96b30271a_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/a/4a49215e09f63680f6bf0999a21ce4a96b30271a_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/a/4a49215e09f63680f6bf0999a21ce4a96b30271a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/a/4a49215e09f63680f6bf0999a21ce4a96b30271a_2_10x10.png"></div>

<h3><a href="https://github.com/jsoref/community-events/runs/5170886359?check_suite_focus=true#step:2:46" target="_blank" rel="noopener nofollow ugc">Baz · jsoref/community-events@00b41a1</a></h3>

  <p>Contribute to jsoref/community-events development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<details><summary>Log Output</summary>
<pre><code class="lang-auto">github.action: __run
github.action_path:
github.action_ref:
github.action_repository:
github.actor: jsoref
github.api_url: https://api.github.com
github.base_ref: main
github.env: /home/runner/work/_temp/_runner_file_commands/set_env_b7cecac1-39d5-423a-838e-fb0a2c1c1831
github.event: Object
github.event_name: pull_request
github.event_path: /home/runner/work/_temp/_github_workflow/event.json
github.graphql_url: https://api.github.com/graphql
github.head_ref: baz
github.job: debug
github.ref: refs/pull/1/merge
github.ref_name: 1/merge
github.ref_protected: false
github.ref_type: branch
github.path: /home/runner/work/_temp/_runner_file_commands/add_path_b7cecac1-39d5-423a-838e-fb0a2c1c1831
github.repository: jsoref/community-events
github.repository_owner: jsoref
github.repositoryUrl: git://github.com/jsoref/community-events.git
github.retention_days: 90
github.run_id: 1835436481
github.run_number: 36
github.run_attempt: 1
github.server_url: https://github.com
github.sha: 064e4c133d0497b3a5d40263caf85945903806ae
github.token: ***
github.workflow: Debug GHA
github.workspace: /home/runner/work/community-events/community-events
</code></pre>
</details>
<p>You have 3 steps, the first one is fairly aware of both pull request and push events.<br>
Your second step assumes everything looks like a push.</p>
<p>The repository I can find is using <code>github.ref</code> which would return results with <code>/merge</code>:</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/ROBOTIS-GIT/Dynamixel2Arduino/blob/2b8b91f8f8b41945d4634f15723ecdec2e862ba1/.github/workflows/dynamixel2arduino_ci.yml#L18">
  <header class="source">

      <a href="https://github.com/ROBOTIS-GIT/Dynamixel2Arduino/blob/2b8b91f8f8b41945d4634f15723ecdec2e862ba1/.github/workflows/dynamixel2arduino_ci.yml#L18" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/ROBOTIS-GIT/Dynamixel2Arduino/blob/2b8b91f8f8b41945d4634f15723ecdec2e862ba1/.github/workflows/dynamixel2arduino_ci.yml#L18" target="_blank" rel="noopener nofollow ugc">ROBOTIS-GIT/Dynamixel2Arduino/blob/2b8b91f8f8b41945d4634f15723ecdec2e862ba1/.github/workflows/dynamixel2arduino_ci.yml#L18</a></h4>



    <pre class="onebox">      <code class="lang-yml">
        <ol class="start lines" start="8" style="counter-reset: li-counter 7 ;">
            <li>
            </li>
<li>jobs:</li>
            <li>  dynamixel2arduino_ci:</li>
            <li>    runs-on: ubuntu-latest</li>
            <li>    strategy:</li>
            <li>      fail-fast: false</li>
            <li>    steps:</li>
            <li>      - uses: actions/checkout@v2</li>
            <li>      - name: Run Arduino CI Script</li>
            <li>        run: |</li>
            <li class="selected">          source &lt;(curl -SLs https://raw.githubusercontent.com/ROBOTIS-GIT/Dynamixel2Arduino/master/install.sh) ${{github.ref}}</li>
            <li>          build_platform opencr</li>
            <li>          build_platform uno</li>
            <li>          build_platform mega2560</li>
            <li>          build_platform mkrzero</li>
            <li>          build_platform portenta</li>
        </ol>
      </code>
    </pre>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I think that if you use <code>base_ref</code> you should be fairly happy, but not you <em>probably</em> don’t want to run it the way you’re running it, since as written in your post, you’re going to run the script twice for the non-pr case, which seems wrong.</p>

**Comment by ROBOTIS-David** [comment]
<p>Very Helpful ! Thank you a lot !!! :grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning::grinning:</p>
